x-pack/plugins/cases/public/components/custom_fields/flyout.test.tsx

```diff
@@ -115,23 +81,136 @@ describe('CustomFieldFlyout ', () => {
   it('calls onCloseFlyout on close', async () => {
     appMockRender.render(<CustomFieldFlyout {...props} />);
 
-    userEvent.click(screen.getByTestId('euiFlyoutCloseButton'));
+    userEvent.click(await screen.findByTestId('euiFlyoutCloseButton'));
 
     await waitFor(() => {
       expect(props.onCloseFlyout).toBeCalled();
     });
   });
 
-  it('renders flyout with data when customField value exist', async () => {
-    appMockRender.render(
-      <CustomFieldFlyout {...{ ...props, customField: customFieldsConfigurationMock[0] }} />
-    );
-
-    expect(await screen.findByTestId('custom-field-label-input')).toHaveAttribute(
-      'value',
-      customFieldsConfigurationMock[0].label
-    );
-    expect(await screen.findByTestId('custom-field-type-selector')).toHaveAttribute('disabled');
-    expect(await screen.findByTestId('text-custom-field-options')).toHaveAttribute('checked');
+  describe('Text custom field', () => {
+    it('calls onSaveField with correct params when a custom field is NOT required', async () => {
+      appMockRender.render(<CustomFieldFlyout {...props} />);
+
+      userEvent.paste(await screen.findByTestId('custom-field-label-input'), 'Summary');
+      userEvent.click(await screen.findByTestId('custom-field-flyout-save'));
+
+      await waitFor(() => {
+        expect(props.onSaveField).toBeCalledWith({
+          key: expect.anything(),
+          label: 'Summary',
+          required: false,
+          type: CustomFieldTypes.TEXT,
+        });
+      });
+    });
+
+    it('calls onSaveField with the correct params when a custom field is required', async () => {
+      appMockRender.render(<CustomFieldFlyout {...props} />);
+
+      userEvent.paste(await screen.findByTestId('custom-field-label-input'), 'Summary');
+      userEvent.click(await screen.findByTestId('text-custom-field-required'));
+      userEvent.paste(await screen.findByTestId('text-custom-field-default-value'), 'Summary');
```

Comment:
super nit: I think is better to have distinct values on each attribute to avoid bugs in the future. For example, setting the `label` to `defaultValue` by mistake.